src/main/kotlin/com/pubnub/api/v2/callbacks/EventEmitter.kt

```diff
@@ -10,12 +9,144 @@ import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResu
 import com.pubnub.api.models.consumer.pubsub.objects.PNObjectEventResult
 
 interface EventEmitter {
-    var onMessage: ((PubNub, PNMessageResult) -> Unit)?
-    var onPresence: ((PubNub, PNPresenceEventResult) -> Unit)?
-    var onSignal: ((PubNub, PNSignalResult) -> Unit)?
-    var onMessageAction: ((PubNub, PNMessageActionResult) -> Unit)?
-    var onObjects: ((PubNub, PNObjectEventResult) -> Unit)?
-    var onFile: ((PubNub, PNFileEventResult) -> Unit)?
+
+    /**
+     * A nullable property that can be set to a function (or lambda expression) to handle incoming message events.
+     * This function is invoked whenever a new message is received, providing a convenient way to process or react to messages.
+     *
+     * Note that this property allows for the assignment of a singular behavior at a time, as any new assignment will override the previous one.
+     * For scenarios requiring multiple behaviors in response to message events, it is advisable
+     * to utilize `com.pubnub.api.PubNub.addListener(listener: EventListener)`.
+     *
+     * To deactivate the current behavior, simply set this property to `null`.
+     *
+     * **Setting a Behavior Example**:
+     * ```
+     * onMessage = { pnMessageResult ->
+     *     println("Received message: ${pnMessageResult.message}")
+     * }
+     * ```
+     * **Removing a Behavior Example**:
+     * ```
+     * onMessage = null
+     * ```
+     */
+    var onMessage: ((PNMessageResult) -> Unit)?
+
+    /**
+     * A nullable property designed to set a function or lambda expression for handling incoming presence events.
+     * This function is called whenever a new presence event occurs, offering an efficient method for tracking presence updates.
+     *
+     * Note that this property allows for the assignment of a singular behavior at a time, as any new assignment will override the previous one.
+     * For scenarios requiring multiple behaviors in response to presence events, it is advisable
+     * to utilize `com.pubnub.api.PubNub.addListener(listener: EventListener)`.
```

Comment:
everywhere you have this, you could do this instead:
```
to utilize [EventEmitter.addListener].
```

because we don't want to point people from using e.g. Subscription.onMessage to PubNub.addListener (which can be implied by your version)